lib/Transforms/Scalar/HoistConstantArray.cpp

```diff
@@ -250,15 +256,15 @@ bool CandidateArray::AllGEPUsersAreValid(GEPOperator *gep) {
 
 // Analyze all uses of the array to see if it qualifes as a constant array.
 // We check the following conditions:
-//  1. Make sure alloca is only used by GEP.
+//  1. Make sure alloca is only used by GEP and lifetime intrinsics.
```

Comment:
I'm open to deleting the intrinsics, but I think they don't do any harm.
They only don't make sense if you assume that constant memory isn't allocated dynamically and can't be freed anyway. That's probably true for any back end, but you could also imagine another optimization phase undoing the hoisting again because it understands more context and can remove the memory alltogether. For instance, in one of the cases mentioned above, the global constant array actually ends up being 1.0f for every element - but further opt phases apparently fail to understand that, the entire loop with dynamic indexing could be removed using this information.